memory/provider.go

```diff
@@ -57,6 +57,16 @@ func NewDBProviderWithOpts(opts ...ProviderOption) sql.MutableDatabaseProvider {
 	return pro
 }
 
+func (pro *DbProvider) WithTableFunctions(fns []sql.TableFunction) (sql.TableFunctionProvider, error) {
```

Comment:
This is so specific to testing that you should define it on the memory implementation directly, and put it in engine_only_tests.go